.github/workflows/differential-shellcheck.yml

```diff
@@ -0,0 +1,28 @@
+name: Differential ShellCheck
+on:
+  push:
+    branches: [ dev, bugfix ]
+  pull_request:
+    branches: [ dev, bugfix ]
+
+permissions:
+  contents: read
+
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+
+    permissions:
+      security-events: write
+
+    steps:
+      - name: Repository checkout
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Differential ShellCheck
+        uses: redhat-plumbers-in-action/differential-shellcheck@v4
```

Comment:
Thanks for the contribution but I think we're going to utilize a different source to run shellcheck